ruby/bindings/InitRubyBindings.cpp

```diff
@@ -368,11 +389,522 @@ module OpenStudio
   end
 end # module OpenStudio
 
-)END";
+)ruby";
 
   openstudio::evalString(ruby_typedef_script);
 }
 
+void initEmbeddedGems() {
+
+  std::string initScript = R"ruby(
+# This is the save puts to use to catch EPIPE. Uses `puts` on the given IO object and safely ignores any Errno::EPIPE
```

Comment:
There are a couple of functions here, that basically do similar things to embedded_help.rb

Reply:
Not going to comment the 500 lines here.